apps/snitch_core/test/domain/order/transitions_test.exs

```diff
@@ -0,0 +1,144 @@
+defmodule Snitch.Domain.Order.TransitionsTest do
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+
+  import Snitch.Factory
+
+  alias BeepBop.Context
+  alias Ecto.Multi
+  alias Snitch.Data.Schema.Order
+  alias Snitch.Domain.Order.Transitions
+
+  @patna %{
+    first_name: "someone",
+    last_name: "enoemos",
+    address_line_1: "BR Ambedkar Chowk",
+    address_line_2: "street",
+    zip_code: "11111",
+    city: "Rajendra Nagar",
+    phone: "1234567890",
+    country_id: nil,
+    state_id: nil
+  }
+
+  setup :states
+
+  describe "associate_address" do
+    setup %{states: [%{country: country} = state]} do
+      patna = %{@patna | country_id: country.id, state_id: state.id}
+
+      [
+        patna: patna,
+        order: struct(insert(:order, user: build(:user)))
+      ]
+    end
+
+    test "fails with bad address", %{patna: patna, order: order} do
+      result =
+        order
+        |> Context.new(
+          state: %{
+            billing_address: patna,
+            shipping_address: %{patna | state_id: nil}
+          }
+        )
+        |> Transitions.associate_address()
+
+      refute result.valid?
+
+      assert %{
+               valid?: false,
+               changes: %{
+                 shipping_address: %{
+                   action: :insert,
+                   valid?: false,
+                   errors: [state_id: {"state is explicitly required for this country", _}]
+                 }
+               }
+             } = result.multi
+    end
+
+    test "with an order that has no addresses", %{patna: patna, order: order} do
+      assert is_nil(order.billing_address) and is_nil(order.shipping_address)
+
+      result =
+        order
+        |> Context.new(state: %{billing_address: patna, shipping_address: patna})
+        |> Transitions.associate_address()
+        |> IO.inspect(label: "associating address")
```

Comment:
Is this needed?

Reply:
Of course not. Just added for debugging the actual input and output.

Reply:
@SagarKarwande does `mix test` worked as we expected in your case?